request_test.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"net/url"
+	"strings"
+	"testing"
+)
+
+func TestParseRequestLine(t *testing.T) {
+	requestLine := "GET /todos HTTP/1.1"
+	expectedMethod := "GET"
+	expectedRequestURI := "/todos"
+	expectedProto := "HTTP/1.1"
+	if actualMethod, actualRequestURI, actualProto, _ := parseRequestLine(requestLine);
+		!(actualMethod == expectedMethod && actualRequestURI == expectedRequestURI && expectedProto == actualProto) {
+		t.Fatalf("Unexpected result:\nGot:\t%s, %s, %s\nExpected:\t%s, %s, %s\n",
+			actualMethod, actualRequestURI, actualProto,
+			expectedMethod, expectedRequestURI, expectedProto)
+	}
+	return
+}
+
+type reqTest struct {
+	Raw     string
+	Req     *Request
+	Error   string
+}
+
+var noError = ""
+
+var reqTests = []reqTest{
+	// Baseline test; All Request fields included for template use
```

Comment:
are all these test setups necessary? does the HTTP test suite require you to handle gzip, deflate encoding, and different types of charsets? instead of creating one test that covers all tests, I would break them down to a smaller one that you need for the test suite to pass. that way the unit test description is understandable.

Reply:
✅ Headers were gutted to remove gzip, deflate encoding, and different types of charsets. I left them in there originally w/ the purpose of testing that my `request reader` could properly convert any header type to one that it can read, but I agree, is misleading -> removed and tests were simplified.